.github/workflows/draw-zmk.yml

```diff
@@ -111,15 +128,21 @@ jobs:
               keymap "${config_arg[@]}" parse -z "$keymap_file" $parse_args >"${{ inputs.output_folder }}/$keyboard.yaml" \
               && keymap "${config_arg[@]}" draw "${{ inputs.output_folder }}/$keyboard.yaml" $draw_args >"${{ inputs.output_folder }}/$keyboard.svg" \
               || echo "ERROR: parsing or drawing failed for $keyboard!"
+              if [[ ${FIRST} -ne 1 ]]; then
+                OUTPUTS="${OUTPUTS},"
+              fi
+              OUTPUTS="${OUTPUTS}\"${{ inputs.output_folder }}/$keyboard.yaml\",\"${{ inputs.output_folder }}/$keyboard.svg\""
           done
+          echo "OUTPUTS=[${OUTPUTS}]" >> $GITHUB_OUTPUT
```

Comment:
While I understand your use case, I am leaning towards keeping this simple and replicating the existing commit behavior `file_pattern: '${{ inputs.output_folder }}/*.svg ${{ inputs.output_folder }}/*.yaml'` for the artifact. Hand-creating the json seems a bit too complex and fragile.

Reply:
It would have been ideal to scope the artifact to only the produced files, but in the absence of a simple solution I could prefer the non-ideal end result.

What about a tempdir-based approach? e.g.
```sh
artifact_dir=${TMPDIR:-/tmp}/keymap-drawer 
echo "artifact_dir=$artifact_dir" >> $GITHUB_ENV
mkdir -p "$artifact_dir"

# use $artifact_dir for outputs

cp "$artifact_dir/*" "${{ inputs.output_folder }}/"

# use steps.draw.outputs.artifact_dir in artifacts step
```

Reply:
Hm, I think that's more convoluted. I read from your comments, that the JSON approach is something you would like to avoid. Since I am new to github workflows, maybe there is another approach, which I did not find yet. Let me research this and get back to you tomorrow. If I cant find a cleaner approach, I would rather stick with your suggestion to replicate the commit behaviour.

Reply:
Hi Cem, as promised I tried various different methods of passing a bash array variable to the output for another step/job, but nothing worked. The only way is to hand over a JSON array. Since the previous construction was a bit unwieldy, I have now come up with a better, more readable way. The output of several runs with multiple keyboard keymaps simulated and also stray svg/yaml files that we (or better I) rather not have in the artifact can be found here: https://github.com/michaelrommel/gh-workflow-test/actions/workflows/test.yaml

I would be happy to apply this method to the PR, if you want to go this route. Otherwise please feel free to revert back to your original suggestion with the shell globbing patterns. For me it is not a problem. Here the simplified code sample:

```
      - name: Draw keymaps
        id: draw
        run: |
          IFS=','
          declare -a DRAWINGS
          for keymap_file in config/*.keymap; do
              keyboard=$(basename -s .keymap "$keymap_file")
              echo "INFO: drawing for $keyboard"
              ... do the processing ...
              DRAWINGS+=(\"images/$keyboard.yaml\" \"images/$keyboard.svg\")
          done
          echo "DRAWINGS=[${DRAWINGS[*]}]" >> $GITHUB_OUTPUT
      - name: Artifact upload
        uses: actions/upload-artifact@v4
        with:
          name: 'drawings'
          path: |
            ${{ join(fromJSON(steps.draw.outputs.DRAWINGS), '
            ') }}
       
```
I hope this new approach finds your approval. Sorry, but the JSON limitation is really sth out of my control.

Have a good weekend!

  Michael.

Reply:
I just applied the suggestion to the PR, but now I cannot run the workflow anymore. Very strange:

first the commandline arguments were interpreted with blanks:

<img width="1173" alt="image" src="https://github.com/caksoylar/keymap-drawer/assets/919935/9e6430aa-dc7f-4f1b-b2d6-057198897aed">

Then I removed that and it gave an error for the `-j` argument. The usage output also differs, as if a different version was installed into the workflow VM:

<img width="737" alt="image" src="https://github.com/caksoylar/keymap-drawer/assets/919935/ac241aaf-dc40-4cd9-9d0f-b4dbb726ab98">

I cannot figure out, what happens here.

Reply:
Oh, my bad, I think the IFS declaration now screws up with the rest of the arrays. sigh. I have to fix that.

Reply:
I have added the changes to the branch. It works now for my zmk-config as expected.

Reply:
That looks better, thanks -- I don't want to keep nitpicking over this in order to respect your time as well. Please add it to the PR if you could and we can merge.